src/index.tsx

```diff
@@ -2,57 +2,168 @@ import React from 'react';
+       >
+        {props.value}
       </button>
     );
-  }
+  
 }
 
-class Board extends React.Component {
+interface BoardState{squares: Array<any>, xIsNext: Boolean}
+interface BoardProp{squares: Array<any>, onClick: (i: number) => void, winner: {letter: string, lines: number[]} | null}
+class Board extends React.Component<BoardProp,BoardState> {
   renderSquare(i: number) {
-    return <Square />;
+    return <Square 
+      value={this.props.squares[i]} 
+      onClick={() => this.props.onClick(i)}
+      inLines={this.props.winner?.lines.includes(i) || false}  
+    />;
   }
+  loopSquare(){
+    let squareArray = []
+    let rowArray = []
+    for (let i = 0 ;i<9; i++){
+      squareArray.push(this.renderSquare(i))
+    }
 
-  render() {
-    const status = 'Next player: X';
+    for(let row=0; row<3; row++){
+        rowArray.push(<div className="board-row">{squareArray.slice((row*3), (row*3)+3)}</div>)    
+    }
+
+    // for(let row=0; row<3; row++){
+    //   let squareArray = []
+    //   for (let i = row*3 ; i<(row*3)+3; i++){
+    //     squareArray.push(this.renderSquare(i))
+    //   }
+    //   rowArray.push(<div className="board-row">{squareArray}</div>)
+    // }
+
+    return rowArray
+   
+  }
+  
 
+  render() {
     return (
       <div>
-        <div className="status">{status}</div>
-        <div className="board-row">
-          {this.renderSquare(0)}
-          {this.renderSquare(1)}
-          {this.renderSquare(2)}
-        </div>
-        <div className="board-row">
-          {this.renderSquare(3)}
-          {this.renderSquare(4)}
-          {this.renderSquare(5)}
-        </div>
-        <div className="board-row">
-          {this.renderSquare(6)}
-          {this.renderSquare(7)}
-          {this.renderSquare(8)}
-        </div>
+          {this.loopSquare()}
       </div>
     );
   }
 }
 
-class Game extends React.Component {
+function calculateWinner(squares: Array<any>){
+  const lines = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+  ];
+  for (let i = 0; i < lines.length; i++) {
+    const [a, b, c] = lines[i];
+    if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
+      return {letter: squares[a], lines: lines[i]} ;
+    }
+  }
+  return null;
+}
+
+interface GameState{xIsNext: Boolean, history: Array<any>, stepNumber: number, coordinates: Array<any>}
+class Game extends React.Component<{}, GameState> {
+  constructor(props: {}){
+    super(props);
+    this.state = {
+      history: [{
+        squares: Array(9).fill(null),
+      }],
+      stepNumber: 0,
+      xIsNext: true,
+      coordinates: [{
+        col: 0,
+        row: 0,
+      }]
+    }
+  }
+
+  handleClick(i: number) {
+    const history = this.state.history;
+    const current = history[this.state.stepNumber];
+    const squares = current.squares.slice();
+    const coordinates = this.state.coordinates
```

Comment:
@meep130 Is this variable being used? If not, please remove it.

Reply:
@yapTim done, please check again